src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/haflows/HaFlowPathsSpec.groovy

```diff
@@ -38,4 +35,30 @@ class HaFlowPathsSpec extends HealthCheckSpecification {
         def actualException = thrown(HttpClientErrorException)
         new HaFlowNotFoundExpectedError(~/HA-flow ${flowId} not found\./).matches(actualException)
     }
+
+    @Tidy
+    def "HA flow main path is not overlapped with protected path"() {
+        assumeTrue(useMultitable, "HA-flow operations require multiTable switch mode")
+        given: "An HA-flow with protected path"
+        def swT = topologyHelper.findSwitchTripletForHaFlowWithProtectedPaths()
+        assumeTrue(swT != null, "These cases cannot be covered on given topology:")
+
+        def haFlowRequest = haFlowHelper.randomHaFlow(swT)
+        haFlowRequest.allocateProtectedPath = true
+        HaFlowExtended haFlow = swT.addHaFlow(haFlowRequest)
+        assert haFlow.allocateProtectedPath
+
+        when: "Request path for HA flow with protected path"
+        def haFlowPaths = haFlow.getAllEntityPaths()
+        assert haFlowPaths.subFlowPaths.protectedPath.size() == 2
+
+        then: "HA flow main and protected paths do not have common ISLs"
+        haFlowPaths.subFlowPaths.each {subFlowPath ->
```

Comment:
What if rewrite loop using `every` method? E.g. like
```
haFlowPaths.subFlowPaths.every {it.forward.getCommonIsls(it.protectedPath.forward).isEmpty() }
```
Also, now, when you have FlowPathModel, you can add method .getCommonIslsWithProtected(), so your assertion would look like 
```
haFlowPaths.subFlowPaths.every {it.getCommonIslsWithProtected().isEmpty() }
```

Reply:
The appropriate method has been added to the FlowPathModel.
As for the usage of each{}, let's keep it as there is no need to iterate through all elements when we find a corrupted one.